packages/worker/src/spot-termination.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	changeMessageVisibility,
+	logger,
+} from '@guardian/transcription-service-backend-common';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { getCurrentReceiptHandle, setInterruptionTime } from './index';
+
+const CHECK_FREQUENCY = 10;
+
+export const checkSpotInterrupt = async (
+	client: SQSClient,
+	queueUrl: string,
+) => {
+	const url = 'http://169.254.169.254/latest/meta-data/spot/instance-action';
+	const result = await fetch(url);
+	if (result.status === 200) {
+		const json = await result.json();
+		if (json.action === 'terminate') {
+			setInterruptionTime(new Date(json.time));
+			logger.warn('Spot instance termination detected');
+			// Interrupt warning occurs 2 minutes before termination
+			const receiptHandle = getCurrentReceiptHandle();
+			if (!receiptHandle) {
+				return;
+			}
+			await changeMessageVisibility(client, queueUrl, receiptHandle, 110);
+			// once the interrupt warning has happened, we don't need to keep checking
+			return;
```

Comment:
non-blocking: `changeMessageVisibility` rethrows exceptions which shouldn't matter because there's nothing left to do. It might be worth either changing its signature or wrapping it in a try / catch anyway. This way if we change this function to do more stuff after `changeMessageVisibility` it'll be guaranteed to happen